src/Tribe/Repositories/Event.php

```diff
@@ -1227,4 +1233,34 @@ public function has_date_filters() {
 
 		return false;
 	}
+
+	/**
+	 * Filters events to include only those that start on a specific date.
+	 *
+	 * This method is a wrapper for the `filter_by_starts_between` one.
+	 *
+	 * @since TBD
+	 *
+	 * @param      int|string|\DateTime $date A date and time timestamp, string or object.
+	 * @param null $timezone The timezone that should be used to filter events, if not passed
+	 *                        the site one will be used. This paramenter will be ignored if the
+	 *                       `$date` parameter is an object.
+	 */
+	public function filter_by_on_date( $date, $timezone = null ) {
+		$timezone = Tribe__Timezones::build_timezone_object( $timezone );
+		$date     = DateTimeImmutable::createFromMutable( Tribe__Date_Utils::build_date_object( $date, $timezone ) );
+
+		$format = 'Y-m-d H:i:s';
+		$prev_day = $date->sub( new DateInterval( 'P1D' ) );
+		$prev_day_end = tribe_end_of_day( $prev_day->format( $format ), $format );
+		// Re-calculate using the cutoff time.
+		$prev_day = Tribe__Date_Utils::build_date_object( $prev_day_end, $timezone );
+		$one_second = new DateInterval( 'PT1S' );
+
```

Comment:
We have to catch errors here with the use of DateTimeImmutable and DateInterval, right?

DateTimeImmutable is PHP 5.5+, DateInterval 5.3+, so even with our PHP minimum version upgrades we can't totally use these free and clear just yet..._I think_

Reply:
Good catch, I've updated the code to remove usages of `DateTimeImmutable` and `DateInterval`